src/components/Sections/SectionsProgress.tsx

```diff
@@ -22,19 +34,79 @@ const SectionsProgress: React.FC<SectionsProgressProps> = ({
     )
   }
 
-  const getProgress = ({ doneRequired, doneNonRequired, totalSum, valid }: ProgressType) => {
+  const getIsStrictSection = (sectionCode: string) =>
+    sectionCode === firstStrictInvalidPage?.sectionCode
+
+  const setIsAfterStrict = (sectionCode: string) => {
+    if (!isAfterStrict)
+      isAfterStrict = firstStrictInvalidPage === null ? true : getIsStrictSection(sectionCode)
+  }
+  const SectionProgress: React.FC<ProgressType> = ({
+    doneRequired,
+    doneNonRequired,
+    totalSum,
+    valid,
+  }) => {
     const totalDone = doneRequired + doneNonRequired
     return totalDone > 0 && totalSum > 0 ? (
-      <Progress
-        style={{ width: 200 }}
-        percent={(100 * totalDone) / totalSum}
-        size="tiny"
-        success={valid}
-        error={!valid}
-      />
+      <Grid.Column style={{ width: 200, paddingRight: '5%' }} floated="right" width={4}>
+        <Progress
+          style={{ width: 200 }}
+          percent={(100 * totalDone) / totalSum}
+          size="tiny"
+          success={valid}
+          error={!valid}
+        />
+      </Grid.Column>
     ) : null
   }
 
+  const SectionAction: React.FC<SectionActionProps> = ({
+    sectionCode,
+    firstStrictInvalidPage,
+    progress,
+  }) => {
+    setIsAfterStrict(sectionCode)
+    return (
+      <Grid.Column style={{ minWidth: 100, padding: 0 }} verticalAlign="middle" width={2}>
+        {canEdit ? (
+          getIsStrictSection(sectionCode) ? (
+            <Button
+              color="blue"
+              onClick={() =>
+                resumeApplication({
+                  sectionCode,
+                  pageNumber: firstStrictInvalidPage?.pageNumber || 1,
+                })
+              }
+            >
+              {strings.BUTTON_APPLICATION_RESUME}
+            </Button>
+          ) : progress?.completed && isAfterStrict ? (
+            <Icon
+              name="pencil square"
+              color="blue"
+              style={{ minWidth: 100 }}
+              onClick={() => resumeApplication({ sectionCode, pageNumber: 1 })}
+            />
+          ) : null
+        ) : changes.state && getIsStrictSection(sectionCode) ? (
```

Comment:
This condition is quite hard to follow.

canEdit ? 
 isStrictSection ? 
     'Resume' : 
     progress?.completed && isAfterStrict ? 
           'Edit' : 
           changes.state && getIsStrictSection(sectionCode) ? 
                changes.label : 
                null


wouldn't this be more clear ?

<Grid.Column style={{ minWidth: 100, padding: 0 }} verticalAlign="middle" width={2}>
  <SectionAction {...{ sectionCode, isBeforeStrict, progress }} />
</Grid.Column>

Inside SectionAction ->

if(!canEdit) return  null

if(isStrictSection) return 'Resume'

if(progress?.completed && isBeforeStrict ) return 'Edit'

if( changes.state) return  changes.label

By the way, I dont' think the 'update' one in the home page can be a button (can't navigate to future pates). Would just be an indicator of how many changes are requested right ?

Reply:
I agree, I generally prefer the "guard clause" pattern more than the complex nested ternary operators.

Reply:
Nice. Thanks for having a thinking about that. Very helpful, will try to implement this solution now

Reply:
Done: de7aac7